rm-datasource/src/main/java/io/seata/rm/datasource/undo/KeywordCheckerFactory.java

```diff
@@ -28,20 +29,31 @@
  */
 public class KeywordCheckerFactory {
 
+    private static volatile Map<String,KeywordChecker> keywordCheckerMap;
+
     /**
      * get keyword checker
      *
      * @param dbType the db type
      * @return keyword checker
      */
     public static KeywordChecker getKeywordChecker(String dbType) {
-        if (dbType.equals(JdbcConstants.MYSQL)) {
-            return MySQLKeywordChecker.getInstance();
-        } else  if (dbType.equals(JdbcConstants.ORACLE)) {
-            return OracleKeywordChecker.getInstance();
-        } else {
-            throw new NotSupportYetException(dbType);
+        if (keywordCheckerMap == null) {
+            synchronized (KeywordCheckerFactory.class) {
+                if (keywordCheckerMap == null) {
+                    Map<String, KeywordChecker> initializedMap = new HashMap<>();
+                    List<KeywordChecker> checkerList = EnhancedServiceLoader.loadAll(KeywordChecker.class);
```

Comment:
Why load all types of keyword checker?I think only one type of keyword is required.
You can use @LoadLevel annotation to load a explict implementation of keywordChecker.
![image](https://user-images.githubusercontent.com/24457941/69939943-011a7480-151c-11ea-9d52-1173e34b995d.png)

Reply:
@xingfudeshi support for multiple data sources

Reply:
@xingfudeshi 
thanks for your advise,it is true that only one checker is used once.

but when a not supported dbType is loaded second or more times,using loadAll(), we can throw unsupported exceptions directly. while using load(), we need to do the following:
(1) obtain the lock(this will also block other thread to abtain the checker of other dbType)
(2) reload all extensions of KeywordChecker,
(3) use reflect to obtain annotation and judge whether the dbtype is supported
(4) throw exception. 

Moreover,getting a explict checker at a time will not bring any performance improvement. reviewing the source code of EnhancedServiceLoader, I found that the difference between loadall() and load() is whether to get comments through reflection and filter by activeName (they both need to load all the extensions)  so I think it will be better to use loadAll()

Reply:
@xingfudeshi 
thanks for your advise,it is true that only one checker is used once.

but when a not supported dbType is loaded second or more times,using loadAll(), we can throw unsupported exceptions directly. while using load(), we need to do the following:
(1) obtain the lock(this will also block other thread to abtain the checker of other dbType)
(2) reload all extensions of KeywordChecker,
(3) use reflect to obtain annotation and judge whether the dbtype is supported
(4) throw exception.

Moreover,getting a explict checker at a time will not bring any performance improvement. reviewing the source code of EnhancedServiceLoader, I found that the difference between loadall() and load() is whether to get annotation through reflection and filter by activeName (they both need to load all the extensions) so I think it will be better to use loadAll()

Reply:
OK.Can we use concurrentHashMap.computeIfAbsent to instead of Double check?
And it seems our SPI loader should be optimized.

Reply:
https://www.yuque.com/cgbcac/aoni5y/uw0n5d
The 50th mission: spi optimize

Reply:
@xingfudeshi 
I am afraid not now,computeIfAbsent is suit for the situation that a single element was load by using double-checked locking,but now we load all the elements at once,the return value of these two situation mismatchs.
we might use it to simplify the code after optimizing the SPI loader.

Reply:
@CharmingRabbit Can you recieve the task 50 to optimize spi?
Not limited to 50 descriptions.

Reply:
@zjinlei 
Yes, I'd like to. Is there a deadline for this task?

Reply:
@CharmingRabbit please add my dingding i8vx5rv